landlab/components/landslides/landslide_probability.py

```diff
@@ -0,0 +1,535 @@
+        self.FS_distribution = self.FS_store
+        count = 0
+        for val in self.FS:                   # find how many FS values <= 1
+            if val <= 1.0:
+                count = count + 1
+        self.FS_L1 = float(count)     # number with unstable FS values (<=1)
+        # probability: No. unstable values/total No. of values (n)
+        self.landslide__probability_of_failure = self.FS_L1/self.n
+
+    def calculate_landslide_probability(self, **kwds):
+
+        """
+        Method creates arrays for output variables then loops through all
+        the core nodes to run the method 'calculate_factor_of_safety.'
+        Some output variables are assigned as fields to nodes. One output
+        parameter is an factor-of-safety distribution at each node.
+
+        Parameters
+        ----------
+        self.landslide__factor_of_safety_distribution: numpy.ndarray([
+            self.grid.number_of_nodes, self.n], dtype=float)
+            This is an output - distribution of factor-of-safety from
+            Monte Carlo simulation (units='None')
+        """
+
+        # Create arrays for data with -9999 as default to store output
+        self.mean_Relative_Wetness = -9999*np.ones(self.grid.number_of_nodes,
+                                                   dtype='float')
+        self.prob_fail = -9999*np.ones(
+            self.grid.number_of_nodes, dtype='float')
+        self.landslide__factor_of_safety_distribution = -9999*np.ones(
+            [self.grid.number_of_nodes, self.n], dtype='float')
+        # Run factor of safety Monte Carlo for all core nodes in domain
+        # i refers to each core node id
+        for i in self.grid.core_nodes:
+            self.calculate_factor_of_safety(i)
+            # Populate storage arrays with calculated values
+            self.mean_Relative_Wetness[i] = self.soil__mean_relative_wetness
+            self.prob_fail[i] = self.landslide__probability_of_failure
+            self.landslide__factor_of_safety_distribution[i] = (
+                self.FS_distribution)
+            # stores FS values from last loop (node)
+        # replace unrealistic values in arrays
+        self.mean_Relative_Wetness[
+            self.mean_Relative_Wetness < 0.] = 0.  # so can't be negative
+        self.prob_fail[self.prob_fail < 0.] = 0.   # can't be negative
+        # assign output fields to nodes
+        self.grid.at_node['soil__mean_relative_wetness'] = (
+            self.mean_Relative_Wetness)
+        self.grid.at_node['landslide__probability_of_failure'] = self.prob_fail
+
+
+    def seed_generator(self, seed=0):
+        """Seed the random-number generator. This method will create the same
+        sequence again by re-seeding with the same value (default value is
+        zero). To create a sequence other than the default, assign non-zero
+        value for seed.
+        """
+        np.random.seed(seed)
+
+
+    def _interpolate_HSD_dict(self):
```

Comment:
Thanks for adding the docstring. However, here and elsewhere I think you need to add more information about the HSD_dict.